File: domain/src/main/java/cn/byronlab/weather/domain/repository/WeatherRepository.java
package cn.byronlab.weather.domain.repository;

import cn.byronlab.weather.domain.model.Weather;
import cn.byronlab.weather.domain.result.DomainResult;

public interface WeatherRepository {

    DomainResult<Weather> getWeather(String cityId, boolean refreshNow);

    DomainResult<Weather> refreshWeather(String cityId);
}

File: domain/src/main/java/cn/byronlab/weather/domain/repository/WeatherRepository.kt
package cn.byronlab.weather.domain.repository

import cn.byronlab.weather.domain.model.Weather
import cn.byronlab.weather.domain.result.DomainResult

interface WeatherRepository {

    suspend fun getWeather(cityId: String, refreshNow: Boolean): DomainResult<Weather>

    suspend fun refreshWeather(cityId: String): DomainResult<Weather>
}
